Added support for "bibliography" directive (#3)
scheunemann/pelican-bib: README.md

Register the plugin folder and add the plugin to the `PLUGINS` variable (in your Pelican config, e.g. `pelicanconf.py`):

```python
PLUGIN_PATHS = ['plugins/pelican-bib', ...]
PLUGINS = ['pelican_bib', ...]
```

## How to Use

This plugin reads a user-specified BibTeX file and populates the context with
a list of publications, ready to be used in your Jinja2 template.

Configuration is simply: You can embed a list of publications _via directive_ 
or a _page template_.

### Option 1: "bibliography" directive

Use the `bibliography` directive in your reST files and pass the path (or multiple paths)
 to the bibliography file (relative to current document or absolute to Pelican content path).

```rst
.. bibliography:: pubs.bib
.. bibliography:: /pages/publications/pubs.bib
.. bibliography:: pubs1.bib pubs2.bib pubs3.bib
```

or provide BibTeX entries as content.

```rst
.. bibliography:: 

  @inproceedings{doe_conf_2020,
    author={Doe, Jane and Ordinary, Joe}, 
    title={The Title of the Work},
    booktitle={Proceedings of the Conference (CONF)},
    year={2020}, pages={to appear}
  }
```

Both input methods (path\[s\] and content) can be combined - entries will then be merged.

> You will be able to find the `publications` variable your template.

Reviewer: ```suggestion
You will be able to use the variable `publications` in your template.
```

  In <span class="bib-btitle"><em>Proceedings of the Conference (CONF)</em></span>, to appear. 
  <span class="bib-address_organization_publisher_date">2020.</span>
</span>
```

Available CSS classes are (according to the `format_*` and `get_*_template` function names of 
[pybtex.style.formatting.unsrt](https://bitbucket.org/pybtex-devs/pybtex/src/master/pybtex/style/formatting/unsrt.py)):

_bib-names_, 
_bib-article_, 
_bib-author_or_editor_, 
_bib-editor_, 
_bib-volume_and_series_, 
_bib-chapter_and_pages_, 
_bib-edition_, 
_bib-title_, 
_bib-btitle_, 
_bib-address_organization_publisher_date_, 
_bib-book_, 
_bib-booklet_, 
_bib-inbook_, 
_bib-incollection_, 
_bib-inproceedings_, 
_bib-manual_, 
_bib-mastersthesis_, 
_bib-misc_, 
_bib-phdthesis_, 
_bib-proceedings_, 
_bib-techreport_, 
_bib-unpublished_, 
_bib-web_refs_, 
_bib-url_, 
_bib-pubmed_, 
_bib-doi_, 
_bib-eprint_.

You can use the `class` option of the `bibliography` directive to change the class attribute name (default is "bibliography" and the bibliography file name) of the publication list's container. E.g.

```rst
.. bibliography:: pubs.bib
   :class: my-custom-bib

> ```

Reviewer: I would re-order the explanation. Also note that I changed to plural regarding the "bibliography file name**s**".

```suggestion
Each bibliography is wrapped in a `div`-container with the default classes "bibliography" and the bibliography file names. Use the `class` option of the `bibliography` directive to change the class attribute name:

```rst
.. bibliography:: pubs.bib
   :class: my-custom-bib
```
```

```

or just use the predifined options:

- `sorting_style`. values: `author_year_title` (default) or `none` (sorting exactly as in the bib file) 
- `abbreviate_names`. values: `True` or `False` (default)
- `name_style`. values: `plain` (default) or `lastfirst`

```rst
.. bibliography:: pubs.bib
   :sorting_style: author_year_title
   :abbreviate_names: True
   :name_style: lastfirst
```

### Split into lists of publications

You can add an extra field to each BibTeX entry. This value of that field is a comma separated list.
These values will become the keys of a list `publications_lists` containing the associated BibTeX entries in your template.

For example, if you want to associate an entry with two different tags (`foo-tag`, `bar-tag`), 
you add the following field to the bib entry:

```bib
@article{
   ...
   tags = {foo-tag, bar-tag}
}
```

In your `pelicanconf.py` you'll need to set:

```python
PUBLICATIONS_SPLIT_BY = 'tags'
```

In your template you can then access these lists with the variables `publications_lists['foo-tag']` and `publications_lists['bar-tag']`.

> You can use the `filter-tags` option of the `bibliography` directive to only render publications containing the given tag:

Reviewer: ```suggestion
You can use the `filter_tag` option of the `bibliography` directive to only render publications containing the given tag:
```

```rst
.. bibliography:: pubs.bib

>    :filter-tag: foo-tag

Reviewer: ```suggestion
   :filter_tag: foo-tag
```

alternative path to the `pybtex_plugins.py` file can be provided via
`PUBLICATIONS_PLUGIN_PATH` in the Pelican config.

## Example templates

First, add the following content to your template file.

Variant A - Open BibTeX entry via JavaScript in new windows:

```jinja
{% macro render_publication(publication) %}
    <li id="{{ publication.key }}">
        {{ publication.text }}

>         [&nbsp;<a href="javascript:window.open().document.write('<pre>{{ publication.bibtex|replace('\n', '\\n')|escape|forceescape }}</pre>');">Bibtex</a>&nbsp;]

Reviewer: This gives output which is not properly formatted, e.g., line-breaks and indents missing. I suggest using the original `disp`lay function (as before).

```suggestion
        [&nbsp;<a href="javascript:disp('{{ publication.bibtex|replace('\n', '\\n')|escape|forceescape }}');">Bibtex</a>&nbsp;]
```

Author: The idea here was to omit a separate `<script>` block and to define all neccessary JavaScript inside the `href` attribute.
What leads to the output not being formatted properly. (or: Which code in the `display` function ensures that the output is properly formatted.)
